src/frontend/src/optimizer/logical_optimization.rs

```diff
@@ -424,15 +424,18 @@ impl LogicalOptimizer {
 
         plan = Self::predicate_pushdown(plan, explain_trace, &ctx);
 
+        // the `OverWindowToTopNRule` need to match the pattern of Proj-Filter-OverWindow so it is
+        // 1. conflict with `ProjectJoinMergeRule`, `AggProjectMergeRule` or other rules
+        // 2. should be after merge the multiple projects
```

Comment:
No, the merge is needed to deal with the Plan
```
Project(x)-Project(x, rank)-Filter(x<=N)-OverWindow()
```
and the OverWindowToTopNRule can only handle
```
Project(x)-Filter(x<=N)-OverWindow()
```

Reply:
I see. Since each rule in `OptimizationStage` applies in an order. Maybe reordering the rule order (placing `OverWindowToTopNRule` at the end) is enough for this case? We don't need `optimize_by_rules_until_fix_point`?

Reply:
It works! thanks for the advice.